### PR TITLE
Added keep scale flag to billboard materials

### PR DESCRIPTION
**Pull request by JFonS**
Adds support for #18292.

### CODE REVIEW

File: scene/resources/material.h [hunk 1]
```diff
@@ -242,7 +243,7 @@ class SpatialMaterial : public Material {
 			uint64_t specular_mode : 2;
 			uint64_t invalid_key : 1;
 			uint64_t deep_parallax : 1;
-			uint64_t billboard_mode : 2;
+			uint64_t billboard_mode : 3;
```
Comment (reduz) [inline scene/resources/material.h]: We'll soon run out of 64 bits :P

**Comment by akien-mga** [comment]
Note: AppVeyor build issue is a false positive.

**Comment by akien-mga** [comment]
Agreed on PR review meeting to merge once rebased.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Now that Godot 3.1 has been released, we don't plan to cherry-pick new features and enhancements to the 3.0 branch, unless there is very strong support in favor of it. Removing cherrypick label for 3.0.

**Pull request merged**